pkg/state/state_skel.go

```diff
@@ -91,32 +91,104 @@ func (s *stateSkel) updateObj(obj *unstructured.Unstructured) error {
 	return nil
 }
 
+// When updating, not all resources need an updated ResourceVersion
+func needToUpdateResourceVersion(kind string) bool {
+	if kind == "SecurityContextConstraints" ||
+		kind == "Service" ||
+		kind == "ServiceMonitor" ||
+		kind == "Route" ||
+		kind == "BuildConfig" ||
+		kind == "ImageStream" ||
+		kind == "PrometheusRule" {
+		return true
+	}
+	return false
+}
+
+func checkNestedFields(found bool, err error) {
+	if !found || err != nil {
+		return errors.Wrap(err, "nested field error")
+	}
+}
+
+func updateResourceVersion(req, found *unstructured.Unstructured) error {
```

Comment:
we update both resourceVersion and clusterIP for service.
down the road it might update some other fields depending on the object kind.

perhaps call it: `updateInternalFields` ?